### PR TITLE
ISLE-v.1.2.0 PR for release

### PR DESCRIPTION
**Pull request by g7morris**
@bseeger When you have a chance, please review this ISLE-v.1.2.0 PR for release. Thank you.

### CODE REVIEW

File: docs/install/install-local-new.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+```bash  
+icg-upstream	https://github.com/Islandora-Collaboration-Group/ISLE.git (fetch)
+icg-upstream	https://github.com/Islandora-Collaboration-Group/ISLE.git (push)
+origin	https://yourgitproviderhere.com/yourinstitutionhere/yourprojectnamehere-isle.git (fetch)
+origin	https://yourgitproviderhere.com/yourinstitutionhere/yourprojectnamehere-isle.git (push)
+```
+
+* Run a git fetch
+  * `git fetch icg-upstream`
```
Comment (bseeger) [inline docs/install/install-local-new.md]: This is unnecessary as the `git pull` does both a fetch and merge.  Do you have reason you like to do it this way?  Thanks!

Comment (g7morris) [reply docs/install/install-local-new.md]: Yeah it is probably overkill but I've found in the past that I've not always gotten the "latest" updates from running `git pull` so I've started to run `git fetch` as a pre-caution. I tend to jump between using git cli and the Sourcetree Mac app. I don't know which process fails me but one isn't consistent in getting branch updates from the remote(s).

File: docker-compose.local.yml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+## Local Environment - Used for Drupal site development, more extensive metadata, Solr and Fedora development work on a local laptop or workstation
+## Updated 2019-08 - Release 1.2.0 (@ 1.2.0)
+
+services:
+  isle-portainer: ## Renamed to avoid conflicts on systems/servers with portainer already running.
+    image: portainer/portainer
+    container_name: isle-portainer-${CONTAINER_SHORT_ID}
+    command: -H unix:///var/run/docker.sock --no-auth
+    env_file:
+      - demo.env
```
Comment (bseeger) [inline docker-compose.local.yml]: is this the correct env file name here, or should it be local.env?

File: docker-compose.staging.yml [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+## Staging Environment - Used for remote (non-local / cloud) hosting of a Staging site. Code here is almost finished but viewed privately prior to pushing to public Production.
+## Updated 2019-08 - Release 1.2.0 (@ 1.2.0)
+
+services:
+  isle-portainer: ## Renamed to avoid conflicts on systems/servers with portainer already running.
+    image: portainer/portainer
+    container_name: isle-portainer-${CONTAINER_SHORT_ID}
+    command: -H unix:///var/run/docker.sock --no-auth
+    env_file:
+      - demo.env
```
Comment (bseeger) [inline docker-compose.staging.yml]: Is this the correct env file here, or should it be local.env?

File: docker-compose.local.yml (hunk 2 above)
Comment (g7morris) [reply docker-compose.local.yml]: Well spotted. This is a typo introduced from a recent diff, apologies. I've fixed it in all files and pushed a new commit.

File: docker-compose.staging.yml (hunk 3 above)
Comment (g7morris) [reply docker-compose.staging.yml]: Well spotted. This is a typo introduced from a recent diff, apologies. I've fixed it in all files and pushed a new commit.

File: docs/update/update.md [hunk 4]
```diff
@@ -4,24 +4,100 @@ Update an existing ISLE installation to install the newest improvements and secu
...
     * `docker-compose down`
+
+* Run a git fetch from the ICG upstream
+  * `git fetch icg-upstream`
+
+* Checkout a new git branch as a precaution for performing the update on your project. This way your `master` branch stays safe and untouched until you have tested thoroughly and are ready to merge in changes from the recent update. The new local branch can be anything. An example is given below.
+  * `git checkout isle-update-numberhere`
+
+* Pull down the ICG ISLE `master` branch into your local project's new `isle-update-numberhere` branch
+  * `git pull icg-upstream master`
```
[the remark this answers is not in the document]
Comment (g7morris) [reply docs/update/update.md]: Yeah it is probably overkill but I've found in the past that I've not always gotten the "latest" updates from running git pull so I've started to run git fetch as a pre-caution. I tend to jump between using git cli and the Sourcetree Mac app. I don't know which process fails me but one isn't consistent in getting branch updates from the remote(s).

File: docs/install/install-staging-new.md [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+
+* Copy the `local_drupal_site_082019.sql` created in Step 1 to the Remote Staging server.
+
+* Import the exported `Local` MySQL database for use in the current `Staging` Drupal site. Refer to your `staging.env` for the usernames and passwords used below.
+  * You can use a MySQL GUI client for this process instead but the command line directions are only included below.
+  * Run `docker ps` to determine the MySQL container name
+  * Shell into your currently running `Staging` MySQL container
+    * `docker exec -it yourmysql-container-name bash`
+  * Import the Local Drupal database. Replace the `DRUPAL_DB` & `DRUPAL_DB_USER` below in the command with the values found in your `staging.env`.
+    * `mysqldump -u DRUPAL_DB_USER -p DRUPAL_DB < local_drupal_site_082019.sql`
```
Comment (bseeger) [inline docs/install/install-staging-new.md]: I believe this needs to be the root user, as the DRUPAL_DB_USER doesn't exist yet.

Comment (g7morris) [reply docs/install/install-staging-new.md]: By default spinning up a blank staging system will give you a blank / empty database that you'll then import the Production (non-ISLE) database into. This action can be preformed by the DRUPAL_DB_USER which will exist once the ISLE container init process has completed on start up. So no the root user doesn't need to be involved here necessarily.

Comment (bseeger) [reply docs/install/install-staging-new.md]: ah, actually the issue is that should be `mysql` not `mysqldump`, I think.

Comment (g7morris) [reply docs/install/install-staging-new.md]: Yes you are right. I'll update and push.

Comment (g7morris) [reply docs/install/install-staging-new.md]: Thanks for spotting this.

Comment (g7morris) [reply docs/install/install-staging-new.md]: Fixed in this commit. https://github.com/Islandora-Collaboration-Group/ISLE/pull/287/commits/a5e759b06ed49af9f0e737f4b501cf5d558bbc06

File: docs/optional-components/tickstack.md [hunk 6]
```diff
@@ -0,0 +1,688 @@
...
+* Shut down all running ISLE containers, Staging first, then production second once Staging is up and running again.
+
+---
+
+#### 01. Docker syslog driver setup
+
+* You will need to use the `sudo` command or become the `root` user for the steps below. Copying this file allows the Docker Daemon to use the `syslog` driver for log files and reporting to the `telegraf` agent.
+
+- Check if you already have an existing `/etc/docker/daemon.json` file. 
+  * If **YES**, add / merge  the contents of `/ISLE/config/proxy/tick/docker/daemon.json` with your current `/etc/docker/daemon.json` file.
```
Comment (bseeger) [inline docs/optional-components/tickstack.md]: this path is incorrect for all the daemon.json paths here: it should be `ISLE/config/tick/docker/daemon.json`

File: docs/optional-components/tickstack.md [hunk 7]
```diff
@@ -0,0 +1,688 @@
...
+
+---
+
+#### 02. Rsyslog setup
+
+* You will need to use the `sudo` command or become the `root` user for the steps below. Copying this file allows the information from the host `syslog` be forwarded to the `telegraf` agent.
+
+- Copy the Telegraf configuration file to the Rsyslog.d directory so Rsyslog will forward logs to the Telegraf agent with correct formatting. Your ISLE project location may be different from the example below.
+  
+  * `sudo cp /ISLE/config/proxy/tick/rsyslog/50-telegraf.conf /etc/rsyslog.d/`
```
Comment (bseeger) [inline docs/optional-components/tickstack.md]: the dir `proxy` should be taken out of this path.

**Comment by bseeger** [comment]
Thank you, Gavin, for all your hard work on this!!

**Comment by g7morris** [comment]
Thank you, @bseeger @McFateM @noahwsmith @dwk2 @shauntru for all of your invaluable work and PRs. This was by no means a solo effort and I'm grateful to be a part of this ISLE team. Big release, lots going on and feels good to have so many talented helpful people on board.

**Pull request merged**